clients/rust/tests/harvest_stake_rewards.rs

```diff
@@ -0,0 +1,551 @@
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &authority],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    // Then the destination account has the rewards.
+
+    let account = get_account!(context, destination);
+    assert_eq!(account.lamports, 2_000_000_000);
+
+    // And the stake account has the updated last seen stake rewards per token.
+    let account = get_account!(context, stake_pda);
+    let stake_account = Stake::from_bytes(account.data.as_ref()).unwrap();
+    assert_eq!(
+        stake_account.last_seen_stake_rewards_per_token,
+        40_000_000 * 1_000_000_000 // 0.04 * 1e9
+    );
+}
+
+#[tokio::test]
+async fn harvest_holder_rewards_with_no_rewards_available() {
+    let mut context = ProgramTest::new("stake_program", paladin_stake::ID, None)
+        .start_with_context()
+        .await;
+
+    // Given a config account with no rewards and 100 staked amount and
+    // a validator's vote account.
+
+    let config = create_config(&mut context).await;
+
+    let mut account = get_account!(context, config);
+    let mut config_account = Config::from_bytes(account.data.as_ref()).unwrap();
+    // "manually" set the total amount delegated
+    config_account.token_amount_delegated = 100;
+    account.data = config_account.try_to_vec().unwrap();
+    context.set_account(&config, &account.into());
+
+    let validator = Pubkey::new_unique();
+    let authority = Keypair::new();
+    let vote = create_vote_account(&mut context, &validator, &authority.pubkey()).await;
+
+    // And a stake account wiht a 50 staked amount.
+
+    let stake_pda = create_stake(&mut context, &validator, &vote, &config).await;
+
+    let mut account = get_account!(context, stake_pda);
+    let mut stake_account = Stake::from_bytes(account.data.as_ref()).unwrap();
+    // "manually" set the amount to 50
+    stake_account.amount = 50;
+
+    account.data = stake_account.try_to_vec().unwrap();
+    context.set_account(&stake_pda, &account.into());
+
+    // When we harvest the holder rewards with no rewards available.
+
+    let destination = Pubkey::new_unique();
+
+    let harvest_holder_rewards_ix = HarvestStakeRewardsBuilder::new()
+        .config(config)
+        .stake(stake_pda)
+        .stake_authority(authority.pubkey())
+        .destination(destination)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[harvest_holder_rewards_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &authority],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    // Then the transaction succeeds but the destination account has no rewards.
+
+    let account = context.banks_client.get_account(destination).await.unwrap();
+    assert!(account.is_none());
+}
+
+#[tokio::test]
+async fn harvest_holder_rewards_after_harvesting() {
```

Comment:
nit: this whole test uses "holder" rewards in a few places, but it's all about stake rewards
```suggestion
async fn harvest_stake_rewards_after_harvesting() {
```

Reply:
Oh, my bad.